app/uk/gov/hmrc/agentservicesaccount/controllers/desiDetails/CheckYourAnswers.scala

```diff
@@ -16,18 +16,79 @@
+class CheckYourAnswers @Inject()(actions: Actions,
+                                         sessionCache: SessionCacheService,
+                                         acaConnector: AgentClientAuthorisationConnector,
+                                         pcodRepository: PendingChangeOfDetailsRepository,
+                                         check_updated_details: check_updated_details,
+                                        )(implicit appConfig: AppConfig,
+                                          cc: MessagesControllerComponents,
+                                          ec: ExecutionContext) extends FrontendController(cc) with I18nSupport with Logging {
 
-  def showPage: Action[AnyContent] = Action.async { _ => Ok("").toFuture }
+  def ifFeatureEnabled(action: => Future[Result]): Future[Result] = {
+    if (appConfig.enableChangeContactDetails) action else Future.successful(NotFound)
+  }
 
-  def onSubmit: Action[AnyContent] = Action.async { _ => Ok("").toFuture }
-}
+  private def ifFeatureEnabledAndNoPendingChanges(action: => Future[Result])(implicit request: AuthRequestWithAgentInfo[_]): Future[Result] = {
+    ifFeatureEnabled {
+      pcodRepository.find(request.agentInfo.arn).flatMap {
+        case None => // no change is pending, we can proceed
+          action
+        case Some(_) => // there is a pending change, further changes are locked. Redirect to the base page
+          Future.successful(Redirect(desiDetails.routes.ContactDetailsController.showCurrentContactDetails))
+      }
+    }
+  }
+
+  def showPage: Action[AnyContent] = actions.authActionCheckSuspend.async { implicit request =>
+    ifFeatureEnabledAndNoPendingChanges {
+      sessionCache.get[AgencyDetails](DRAFT_NEW_CONTACT_DETAILS).map {
+        case Some(updatedDetails) => Ok(check_updated_details(updatedDetails, request.agentInfo.isAdmin))
+        case None => Redirect(desiDetails.routes.ContactDetailsController.showCurrentContactDetails)
+      }
+    }
+  }
 
+  def onSubmit: Action[AnyContent] = actions.authActionCheckSuspend.async { implicit request =>
+    ifFeatureEnabledAndNoPendingChanges {
+      val arn = request.agentInfo.arn
+      sessionCache.get[AgencyDetails](DRAFT_NEW_CONTACT_DETAILS).flatMap {
+        case None => // graceful redirect in case of expired session data etc.
+          Future.successful(Redirect(desiDetails.routes.ContactDetailsController.showCurrentContactDetails))
+        case Some(newContactDetails) => for {
+          oldContactDetails <- CurrentAgencyDetails.get(acaConnector)
+          pendingChange = PendingChangeOfDetails(
+            arn = arn,
+            oldDetails = oldContactDetails,
+            newDetails = newContactDetails,
+            timeSubmitted = Instant.now()
+          )
+          //
+          // TODO actual connector call to submit the details goes here...
+          //
```

Comment:
Is this TODO still needed?

Reply:
Yes this was here prior to this ticket. It is presumably part of a separate ticket to get this done